flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -1936,24 +1935,35 @@ static void genStandaloneDistribute(lower::AbstractConverter &converter,
   genDistributeClauses(converter, semaCtx, stmtCtx, item->clauses, loc,
                        distributeClauseOps);
 
-  // TODO: Support delayed privatization.
   DataSharingProcessor dsp(converter, semaCtx, item->clauses, eval,
                            /*shouldCollectPreDeterminedSymbols=*/true,
-                           /*useDelayedPrivatization=*/false, &symTable);
-  dsp.processStep1();
+                           enableDelayedPrivatizationStaging, &symTable);
+  mlir::omp::PrivateClauseOps privateClauseOps;
+  dsp.processStep1(&privateClauseOps);
```

Comment:
I think we can pass `distributeClauseOps` directly to `dsp.processStep1()`. That way we also avoid the need for calling `setPrivateSymsAttr` and appending to `getPrivateVarsMutable` below.

Reply:
Ah, right, good idea. Done!